src/genpy/generator.py

```diff
@@ -870,60 +878,56 @@ def msg_generator(msg_context, spec, search_path):
         yield ''
 
     if len(spec_names):
-        yield "  __slots__ = ['"+"','".join(spec_names)+"']"
-        yield "  _slot_types = ['"+"','".join(spec.types)+"']"
+        yield "  __slots__: List[str] = ['"+"','".join(spec_names)+"']"
+        yield "  _slot_types: List[str] = ['"+"','".join(spec.types)+"']"
     else:
-        yield '  __slots__ = []'
-        yield '  _slot_types = []'
+        yield '  __slots__: List[str] = []'
+        yield '  _slot_types: List[str] = []'
 
-    yield """
-  def __init__(self, *args, **kwds):
+    yield f"""
+  def __init__(self, {', '.join([f'{spec_name}: {format_spec_type_hint} = None' for spec_name, spec_type, format_spec_type_hint in fields])}):
     \"\"\"
     Constructor. Any message fields that are implicitly/explicitly
     set to None will be assigned a default value. The recommend
     use is keyword arguments as this is more robust to future message
     changes.  You cannot mix in-order arguments and keyword arguments.
 
     The available fields are:
-       %s
+      {','.join(spec_names)}
 
     :param args: complete set of field values, in .msg order
     :param kwds: use keyword arguments corresponding to message field names
     to set specific fields.
     \"\"\"
-    if args or kwds:
-      super(%s, self).__init__(*args, **kwds)""" % (','.join(spec_names), name)
-
+    super({name}, self).__init__(**{fields_dict})"""
+    # ^ TODO the above should use defualt values not = None. LEaving as is for the moment until everything is working
```

Comment:
```suggestion
    # ^ TODO the above should use default values not = None. Leaving as is for the moment until everything is working
```
Is this relevant anymore?